File: lib/firebot/service.rb
module Firebot
  class Service < Client
    def email(options = {})
      response = HTTParty.post(
        endpoint("email"), query: options, headers: headers
      )
      json = JSON.parse response.body
      json["errors"] ? json["errors"] : true
    end

    private

    def endpoint(service)
      [api_url, api_version, "services", service].join "/"
    end
  end
end

File: app/assets/javascripts/firebot/service.js
var Service = function() {
  var client = FirebotClient;

  endpoint = function(resource) {
    return client.apiUrl + '/' + client.apiVersion + '/services/' + resource;
  };

  this.email = function(data, callback) {
    $.ajax({
      data: data,
      headers: client.headers,
      method: 'post',
      url: endpoint('email'),
      success: function(response) {
        if (callback) {
          callback(response);
        }
      }
    });
  };
}

window.FirebotService = new Service();
